mtgjson5/price_builder.py

```diff
@@ -252,3 +297,159 @@ def build_prices(self) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+
+        # Prune old v2 records
+        LOGGER.info("Pruning v2 price archive")
+        self._prune_v2_prices_archive(archive_prices_v2)
+
+        # Write and upload v2 archive
+        LOGGER.info("Compressing v2 price data")
+        local_zip_file_v2 = constants.CACHE_PATH.joinpath(bucket_object_path_v2)
+        self.write_price_archive_data(local_zip_file_v2, archive_prices_v2.to_json())
+
+        LOGGER.info("Uploading v2 price data")
+        MtgjsonS3Handler().upload_file(
+            str(local_zip_file_v2), bucket_name, bucket_object_path_v2
+        )
+        local_zip_file_v2.unlink()
+
+        return archive_prices_v2, today_prices_v2
+
+    def _reconstruct_v2_container_from_dict(
+        self, archive_dict: Dict[str, Any]
+    ) -> MtgjsonPricesV2Container:
+        """
+        Reconstruct a MtgjsonPricesV2Container from archived dict format.
+        Archive format is {provider: [record_dicts...]}
+
+        :param archive_dict: Downloaded archive data
+        :return: Reconstructed container
+        """
+        from .classes.mtgjson_prices_v2 import MtgjsonPricesRecordV2
+
+        container = MtgjsonPricesV2Container()
+        if not archive_dict:
+            return container
+
+        LOGGER.info("Loading existing v2 archive data")
+        # Archive is {provider: [records...]} format
+        for provider, records_list in archive_dict.items():
+            for record_dict in records_list:
+                # Reconstruct MtgjsonPricesRecordV2 from dict
+                record = MtgjsonPricesRecordV2(
+                    provider=record_dict.get("provider", provider),
+                    treatment=record_dict["treatment"],
+                    currency=record_dict["currency"],
+                    price_value=record_dict["priceValue"],
+                    price_variant=record_dict["priceVariant"],
+                    uuid=record_dict["uuid"],
+                    platform=record_dict["platform"],
+                    price_type=record_dict["priceType"],
+                    date=record_dict["date"],
+                    subtype=record_dict.get("subtype"),
+                )
+                container.add_record(record)
+
+        return container
+
+    def _merge_v2_prices(
+        self,
+        archive: MtgjsonPricesV2Container,
+        today: MtgjsonPricesV2Container,
+    ) -> None:
+        """
+        Merge today's v2 price records into the archive.
+        Adds all of today's records to the archive (records are timestamped).
+
+        :param archive: Archive container to merge into
+        :param today: Today's prices to merge from
+        """
+        # Simply add all today's records to archive
+        # Each record is timestamped with today's date, so we preserve history
+        # pylint: disable=import-outside-toplevel
+        from .classes.mtgjson_prices_v2 import MtgjsonPricesRecordV2
+
+        today_serialized = today.to_json()
+        for _, records_list in today_serialized.items():
```

Comment:
I like the general idea of the method, but again creating these records could be simpler